ocs_ci/ocs/uninstall.py

```diff
@@ -212,57 +198,93 @@ def uninstall_ocs():
             .get("storageClassName")
         )
 
+    cleanup_policy = (
+        storage_cluster.get()
+        .get("metadata")
+        .get("annotations")
+        .get("uninstall.ocs.openshift.io/cleanup-policy")
+    )
+
     log.info("Deleting storageCluster object")
     storage_cluster.delete(resource_name=constants.DEFAULT_CLUSTERNAME)
 
+    if cleanup_policy == "delete":
+        log.info("Cleanup policy set to delete. checking cleanup pods")
+        cleanup_pods = [
+            pod for pod in get_all_pods() if "cluster-cleanup-job" in pod.name
+        ]
+        for pod in cleanup_pods:
+            while pod.get().get("status").get("phase") != "Succeeded":
+                log.info(f"waiting for cleanup pod {pod.name} to complete")
+                TimeoutSampler(timeout=10, sleep=30)
+            log.info(f"Cleanup pod {pod.name} completed successfully ")
+        # no need to confirm var/vib/rook was deleted from nodes if all cleanup pods are completed.
+    else:
+        log.info("Cleanup policy set to retain. skipping nodes cleanup")
+
+    log.info("Deleting openshift-storage namespace")
+    ocp_obj.delete_project(constants.OPENSHIFT_STORAGE_NAMESPACE)
+    ocp_obj.wait_for_delete(constants.OPENSHIFT_STORAGE_NAMESPACE)
+    switch_to_project(constants.DEFAULT_NAMESPACE)
+
+    # step 10: TODO remove crypto from nodes.
+    """for node in storage_node_list:
+        log.info(f"removing encryption from {node}")
+        ocp_obj.exec_oc_debug_cmd(node=node, cmd_list=[])"""
```

Comment:
still needed?

Reply:
yes. will be used after implementing the encryption function.
will change what's inside the comment